pkg/install/install_test.go

```diff
@@ -120,3 +124,60 @@ func TestDeployARMTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestAddResourceProviderVersion(t *testing.T) {
+	ctx := context.Background()
+
+	clusterdoc := &api.OpenShiftClusterDocument{
+		Key: "test",
+		OpenShiftCluster: &api.OpenShiftCluster{
+			Properties: api.OpenShiftClusterProperties{},
+		},
+	}
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	// The original, as-in-database version of clusterdoc
+	databaseDoc, _ := json.Marshal(clusterdoc)
```

Comment:
Please check errors on all calls (e.g. here and the json.Unmarshal calls below).  Yes, it adds verbosity, but we do it as standard everywhere in our codebase; I believe keeping uniformity and removing the mental overhead of "is this a place where we're checking errors or not?" is worth it.